snuba/datasets/groups.py

```diff
@@ -34,17 +35,22 @@ class Groups(TimeSeriesDataset):
 
     def __init__(self) -> None:
         self.__grouped_message = get_dataset("groupedmessage")
+        groupedmessage_source = self.__grouped_message \
+            .get_dataset_schemas() \
+            .get_read_schema() \
+            .get_data_source()
+        assert isinstance(groupedmessage_source, TableSource)
```

Comment:
The TableJoinNode (below) wants a TableSource not a RelationalSource as expected.
In theory I should be building the schema here with the right type, but since we have groupedmessage, events and the joined one coexisting, I am reusing the schema already defined in groupedmessage and events, thus the cast. 
I can refactor this to have both the single table datasets and the group depending on a schema defined elsewhere but this requires a much bigger refactoring.
I think it is fair for a generic dataset to return RelationalSource and for TableJoinNode to require a TableSource.